packages/semaphore-group-pcd/src/SemaphoreGroupPCD.ts

```diff
@@ -96,28 +97,37 @@ export async function prove(
   args: SemaphoreGroupPCDArgs
 ): Promise<SemaphoreGroupPCD> {
   if (!initArgs) {
-    throw new Error("cannot make group proof: init has not been called yet");
+    throw new Error("Cannot make group proof: init has not been called yet");
   }
 
   const serializedIdentityPCD = args.identity.value?.pcd;
   if (!serializedIdentityPCD) {
-    throw new Error("cannot make group proof: missing semaphore identity PCD");
+    throw new Error("Cannot make group proof: missing semaphore identity PCD");
   }
   const identityPCD = await SemaphoreIdentityPCDPackage.deserialize(
     serializedIdentityPCD
   );
 
   const serializedGroup = args.group.value;
   if (!serializedGroup) {
-    throw new Error("cannot make group proof: missing semaphore group");
+    throw new Error("Cannot make group proof: missing semaphore group");
   }
 
   if (!args.externalNullifier.value) {
-    throw new Error("cannot make group proof: missing externalNullifier");
+    throw new Error("Cannot make group proof: missing externalNullifier");
   }
 
   if (!args.signal.value) {
-    throw new Error("cannot make group proof: missing signal");
+    throw new Error("Cannot make group proof: missing signal");
+  }
+
+  // restrict the group signature from having the same nullifier as the static
+  // signature CPD externalNullifier, which would break anonymity of users if the
+  // adversary also has access to a specific signature
```

Comment:
I think some more explanation would be nice here. Imagine the reader needs like 2 more levels of explanation.